src/main/java/com/mparticle/kits/SkyhookKit.java

```diff
@@ -0,0 +1,170 @@
+package com.mparticle.kits;
+
+import android.Manifest;
+import android.app.Activity;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+
+import com.mparticle.internal.MPUtility;
+import com.skyhookwireless.accelerator.AcceleratorClient;
+import com.skyhookwireless.accelerator.AcceleratorStatusCodes;
+
+import java.util.List;
+import java.util.Map;
+
+public class SkyhookKit
+    extends KitIntegration
+    implements KitIntegration.ActivityListener,
+               AcceleratorClient.ConnectionCallbacks,
+               AcceleratorClient.OnConnectionFailedListener,
+               AcceleratorClient.OnRegisterForCampaignMonitoringResultListener,
+               AcceleratorClient.OnStartCampaignMonitoringResultListener,
+               AcceleratorClient.OnStopCampaignMonitoringResultListener {
+
+    private static final String API_KEY = "apiKey";
+
+    private AcceleratorClient _client;
+    private boolean _isInitialized;
+
+    @Override
+    public String getName() {
+        return "Skyhook";
+    }
+
+    @Override
+    protected List<ReportingMessage> onKitCreate(final Map<String, String> settings,
+                                                 final Context context) {
+        final String apiKey = settings.get(API_KEY);
+        new SkyhookPreferences(context).setApiKey(apiKey);
+
+        _client = new AcceleratorClient(context, apiKey, this, this);
+        SkyhookLog.i("Accelerator SDK v" + _client.getVersion());
+        initialize();
+        return null;
+    }
+
+    @Override
+    protected void onKitDestroy() {
+        _client.stopMonitoringForAllCampaigns(this);
```

Comment:
The check in the boot receiver was for the case when `onKitCreate` was never called at all. When `onKitDestroy` is called, it stops monitoring and persists the state, so when the boot receiver is called next time it would just read the state ("monitoring is off") and stop.

Also, I think that immediately after calling `onKitDestroy` the kit manager (`KitManagerImpl`) will also clear the kit preferences in `onKitCleanup` (because `SkyhookPreferences` uses the kit preference file - `"mp::kit::<KIT_ID>"`).

However, I agree it is cleaner to null the API key out in `onKitDestroy` explicitly. I'll make the change.